File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private List<Integer> result;

    public List<Integer> preorderTraversal(TreeNode root) {
        result = new ArrayList<Integer>();

        preorder(root);
        return result;
    }

    private void preorder(TreeNode node){
        if(node == null){
            return;
        }
        result.add(node.val);
        preorder(node.left);
        preorder(node.right);
    }
}

File: Data Structures & Algorithms/is-palindrome/submission-0.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        left = 0
        right = len(s) - 1

        def check(c):
            if (c >= 'A' and c <='Z') or (c >='a' and c<='z') or (c>='0'and c<='9'):
                return True
            return False

        while left < right:
            while left < right and not check(s[left]):
                left += 1
            while left < right and not check(s[right]):
                right -= 1
            if left >= right:
                break
            if s[left].lower() != s[right].lower():
                return False
            left += 1
            right -= 1
        
        return True
